validator/rules/use_matches.go

```diff
@@ -0,0 +1,44 @@
+package rules
+
+import (
+	"errors"
+	"regexp"
+
+	"github.com/aerogear/charmil/validator"
+	"github.com/spf13/cobra"
+)
+
+// define errors for the UseMatches Rule
+var (
+	ErrInvalidRegexp = errors.New("invalid regexp")
+	ErrRegexMismatch = errors.New("regexp mismatch")
+)
+
+var UseMatchesRule = "USE_MATCHES_RULE"
+
+// UseMatches defines Regexp to be compared
+type UseMatches struct {
+	Regexp string `json:"Regexp"`
+}
+
+// Validate is a method of type Rule Interface
+// which returns validation errors
+func (u *UseMatches) Validate(cmd *cobra.Command) []validator.ValidationError {
+	return u.validateUse(cmd)
+}
+
+// validateUse compares the regexp with Use attribute
+func (u *UseMatches) validateUse(cmd *cobra.Command) []validator.ValidationError {
```

Comment:
If we have more then one thing to do in our `Validate` function it is good to pass them off to new ones but this `validateUse` just seems unnecessary for now.

Reply:
let me change that in all rules, if required?

Reply:
Sure, I can see in the `lengthRule` and `mustExistRule` you have so I think it would be best to quickly change them.

Reply:
done